server/sonar-server/src/main/java/org/sonar/server/settings/ws/ResetAction.java

```diff
@@ -0,0 +1,124 @@
+
+public class ResetAction implements SettingsWsAction {
+
+  private final DbClient dbClient;
+  private final UserSession userSession;
+  private final ComponentFinder componentFinder;
+  private final PropertyDefinitions definitions;
+
+  public ResetAction(DbClient dbClient, ComponentFinder componentFinder, UserSession userSession, PropertyDefinitions definitions) {
+    this.dbClient = dbClient;
+    this.userSession = userSession;
+    this.componentFinder = componentFinder;
+    this.definitions = definitions;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context.createAction(ACTION_RESET)
+      .setDescription("Remove a setting value.<br>" +
+        "Either '%s' or '%s' can be provided, not both.<br> " +
+        "Requires one of the following permissions: " +
+        "<ul>" +
+        "<li>'Administer System'</li>" +
+        "<li>'Administer' rights on the specified component</li>" +
+        "</ul>", PARAM_COMPONENT_ID, PARAM_COMPONENT_KEY)
+      .setSince("6.1")
+      .setPost(true)
+      .setHandler(this);
+
+    action.createParam(PARAM_KEY)
+      .setDescription("Setting key")
+      .setExampleValue("sonar.links.scm")
+      .setRequired(true);
+    addComponentParameters(action);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    DbSession dbSession = dbClient.openSession(false);
+    try {
+      ResetRequest resetRequest = toWsRequest(request);
+      Optional<ComponentDto> component = getComponent(dbSession, resetRequest);
+      checkPermissions(component);
+
+      PropertyDefinition definition = definitions.get(resetRequest.getKey());
+      String key = definition != null ? definition.key() : resetRequest.getKey();
+      if (component.isPresent()) {
+        dbClient.propertiesDao().deleteProjectProperty(key, component.get().getId(), dbSession);
+      } else {
+        dbClient.propertiesDao().deleteGlobalProperty(key, dbSession);
+      }
+      dbSession.commit();
+      response.noContent();
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private static ResetRequest toWsRequest(Request request) {
+    return ResetRequest.builder()
+      .setKey(request.mandatoryParam(PARAM_KEY))
+      .setComponentId(request.param(PARAM_COMPONENT_ID))
+      .setComponentKey(request.param(PARAM_COMPONENT_KEY))
+      .build();
+  }
+
+  private Optional<ComponentDto> getComponent(DbSession dbSession, ResetRequest request) {
+    if (request.getComponentId() == null && request.getComponentKey() == null) {
+      return Optional.empty();
+    }
+    ComponentDto project = componentFinder.getByUuidOrKey(dbSession, request.getComponentId(), request.getComponentKey(), ComponentFinder.ParamNames.COMPONENT_ID_AND_KEY);
+    return Optional.of(project);
+  }
+
+  private void checkPermissions(Optional<ComponentDto> component) {
```

Comment:
@benzonico I think Optional should be allowed as a parameter in **private** methods.